From db54a4e90883ae0f3d841ea06e88be366ba08025 Mon Sep 17 00:00:00 2001
From: Amelia Crate <amelia.crate@chainguard.dev>
Date: Wed, 18 Jun 2025 15:23:58 -0700
Subject: [PATCH] crypto: use prebuilt jitterentropy and sha3 objects

Only the changes at the top of the file referencing jent_sources are necessary,
but guarding the CFLAGS stuff is nice because jitterentropy emits #error if
compiled with optimizations, so this is a small guardrail against unintentional
rebuilding of jitterentropy.
---
 crypto/Makefile | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/crypto/Makefile b/crypto/Makefile
index 017df3a2e..19c4e8b22 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -3,6 +3,16 @@
 # Cryptographic API
 #
 
+jent_sources = jitterentropy jitterentropy-kcapi jitterentropy-testing sha3_generic
+# Use 'make REBUILD_JITTERENTROPY=1' to ignore prebuilt objects
+ifdef REBUILD_JITTERENTROPY
+$(foreach jent_src,$(jent_sources),$(obj)/$(jent_src).c): FORCE
+else
+$(foreach jent_src,$(jent_sources),$(obj)/$(jent_src).o): $(subst, $(obj),$(src),$(@))
+	$(call if-changed,copy)
+endif
+
+
 obj-$(CONFIG_CRYPTO) += crypto.o
 crypto-y := api.o cipher.o
 
@@ -169,9 +179,11 @@ obj-$(CONFIG_CRYPTO_RNG2) += rng.o
 obj-$(CONFIG_CRYPTO_ANSI_CPRNG) += ansi_cprng.o
 obj-$(CONFIG_CRYPTO_DRBG) += drbg.o
 obj-$(CONFIG_CRYPTO_JITTERENTROPY) += jitterentropy_rng.o
+ifdef REBUILD_JITTERENTROPY
 CFLAGS_jitterentropy.o = -O0
 KASAN_SANITIZE_jitterentropy.o = n
 UBSAN_SANITIZE_jitterentropy.o = n
+endif
 jitterentropy_rng-y := jitterentropy.o jitterentropy-kcapi.o
 obj-$(CONFIG_CRYPTO_JITTERENTROPY_TESTINTERFACE) += jitterentropy-testing.o
 obj-$(CONFIG_CRYPTO_BENCHMARK) += tcrypt.o
-- 
2.43.0

